Sources/StreamChat/WebSocketClient/Events/MessageEvents_Tests.swift

```diff
@@ -67,3 +68,121 @@ class MessageEvents_Tests: XCTestCase {
+        
+        // For message to be received, we need to have channel:
+        try client.databaseContainer.createChannel(
+            cid: .init(type: .messaging, id: "general"),
+            withMessages: true,
+            withQuery: false
+        )
+        
+        let unwrappedEvent = try XCTUnwrap(event)
+        client.eventNotificationCenter.process(unwrappedEvent)
+
+        AssertAsync {
+            Assert.willNotBeNil(self.client.databaseContainer.viewContext.message(id: "1ff9f6d0-df70-4703-aef0-379f95ad7366"))
+        }
+    }
+    
+    func test_MessageUpdatedEventPayload_isHandled() throws {
+        let json = XCTestCase.mockData(fromFile: "MessageUpdated")
+        let event = try eventDecoder.decode(from: json) as? MessageUpdatedEvent
+        
+        // For message to be received, we need to have channel:
+        try client.databaseContainer.createChannel(
+            cid: .init(type: .messaging, id: "general"),
+            withMessages: true,
+            withQuery: false
+        )
+    
+        let lastUpdateMessageTime: Date = .unique
+        
+        try client.databaseContainer.createMessage(
+            id: "1ff9f6d0-df70-4703-aef0-379f95ad7366",
+            updatedAt: lastUpdateMessageTime,
+            type: .regular
+        )
+        
+        XCTAssertEqual(
+            client.databaseContainer.viewContext.message(id: "1ff9f6d0-df70-4703-aef0-379f95ad7366")?.updatedAt,
+            lastUpdateMessageTime
+        )
+        
+        let unwrappedUpdate = try XCTUnwrap(event)
+        client.eventNotificationCenter.process(unwrappedUpdate)
+        
+        AssertAsync {
+            Assert.willBeEqual(
+                self.client.databaseContainer.viewContext.message(id: "1ff9f6d0-df70-4703-aef0-379f95ad7366")?.updatedAt
+                    .description,
+                "2020-07-17 13:46:10 +0000"
+            )
+        }
+    }
+    
+    func test_MessageDeletedEventPayload_isHandled() throws {
+        let updateJSON = XCTestCase.mockData(fromFile: "MessageDeleted")
+        let updateMessageEvent = try eventDecoder.decode(from: updateJSON) as? MessageDeletedEvent
+        
+        // For message to be received, we need to have channel:
+        try client.databaseContainer.createChannel(
+            cid: .init(type: .messaging, id: "general"),
+            withMessages: true,
+            withQuery: false
+        )
+        
+        try client.databaseContainer.createMessage(id: "1ff9f6d0-df70-4703-aef0-379f95ad7366", type: .regular)
+        XCTAssertNotNil(client.databaseContainer.viewContext.message(id: "1ff9f6d0-df70-4703-aef0-379f95ad7366"))
+        
+        let unwrappedEvent = try XCTUnwrap(updateMessageEvent)
+        client.eventNotificationCenter.process(unwrappedEvent)
+        
+        AssertAsync {
+            Assert.willNotBeNil(self.client.databaseContainer.viewContext.message(id: "1ff9f6d0-df70-4703-aef0-379f95ad7366"))
+        }
```

Comment:
We should check for `deletedAt` value here

Reply:
Added.